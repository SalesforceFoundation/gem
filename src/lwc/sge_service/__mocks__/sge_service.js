
const SUCCESS = {
    "sObjectName":"Opportunity",
    "sections":[
        {
            "key":"1460808361",
            "label":"Test Section",
            "columns":[
                {
                    "key":"816643058",
                    "fields":[
                        {
                            "typeName":"STRING",
                            "required":true,
                            "name":"Test_Text_Field__c",
                            "label":"Test Text Field",
                            "helpText":null
                        },
                        {
                            "typeName":"PICKLIST",
                            "required":false,
                            "name":"Test_Picklist__c",
                            "label":"Test Picklist",
                            "helpText":null
                        },
                        {
                            "typeName":"REFERENCE",
                            "required":false,
                            "name":"Test_Lookup_Field__c",
                            "label":"Test Lookup Field",
                            "helpText":null
                        },
                        {
                            "typeName":"TEXTAREA",
                            "required":false,
                            "name":"Test_Rich_Text__c",
                            "label":"Test Rich Text",
                            "helpText":null
                        },
                        {
                            "typeName":"PHONE",
                            "required":false,
                            "name":"Test_Phone_Field__c",
                            "label":"Test Phone Field",
                            "helpText":null
                        }
                    ]
                },
                {
                    "key":"211909545",
                    "fields":[
                        {
                            "typeName":"DATE",
                            "required":false,
                            "name":"Test_Date_Field__c",
                            "label":"Test Date Field",
                            "helpText":null
                        },
                        {
                            "typeName":"TEXTAREA",
                            "required":false,
                            "name":"Test_Long_Text_Area__c",
                            "label":"Test Long Text Area",
                            "helpText":null
                        },
                        {
                            "typeName":"PERCENT",
                            "required":false,
                            "name":"Test_Percent_Field__c",
                            "label":"Test Percent Field",
                            "helpText":null
                        },
                        {
                            "typeName":"BOOLEAN",
                            "required":false,
                            "name":"Test_Checkbox_Field__c",
                            "label":"Test Checkbox Field",
                            "helpText":null
                        },
                        {
                            "typeName":"DOUBLE",
                            "required":false,
                            "name":"Test_Number_Field__c",
                            "label":"Test Number Field",
                            "helpText":null
                        }
                    ]
                }
            ]
        }
    ]
};

// resolve to success by default
export const getOpportunityLayout = jest.fn().mockResolvedValue(SUCCESS);