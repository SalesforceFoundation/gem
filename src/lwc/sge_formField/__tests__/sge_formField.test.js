import { createElement } from 'lwc';
import SGE_FormField from 'c/sge_formField';
import { getOpportunityLayout } from 'c/sge_service';

jest.mock('c/sge_service');

describe('c-sge_form-field', () => {

    afterEach(clearDOM);

    it('should load', () => {
        const element = createElement('c-sge_form-field', { is: SGE_FormField });
        element.sobject = {};
        document.body.appendChild(element);
    });

    it('should return a field object with a name and value', () => {
        const element = createElement('c-sge_form-field', { is: SGE_FormField });
        element.sobject = {};
        element.field = {
            'typeName': 'DOUBLE',
            'required': false,
            'name': 'Test_Number_Field__c',
            'label': 'Test Number Field',
            'helpText': null
        };

        document.body.appendChild(element);

        // set value on the lightning-input-field
        const field = element.shadowRoot.querySelector('lightning-input-field');
        field.value = 10.00;

        return flushPromises().then(() => {
            // check for updated value on the sge_form-field component
            const value = element.fieldObject;
            expect(value).toMatchSnapshot();
        });
    });

    it('should work when the value is undefined', () => {
        const element = createElement('c-sge_form-field', { is: SGE_FormField });
        element.sobject = {};
        element.field = {
            'typeName': 'DOUBLE',
            'required': false,
            'name': 'Test_Number_Field__c',
            'label': 'Test Number Field',
            'helpText': null
        };

        document.body.appendChild(element);

        return flushPromises().then(() => {
            // check for updated value on the sge_form-field component
            const value = element.fieldObject;
            expect(value).toMatchSnapshot();
        });
    });

    it('should load with matcing donation info', () => {
        const element = createElement('c-sge_form-field', { is: SGE_FormField });
        element.sobject = {
            'Test_Number_Field__c': 123
        };
        element.field = {
            'typeName': 'DOUBLE',
            'required': false,
            'name': 'Test_Number_Field__c',
            'label': 'Test Number Field',
            'helpText': null
        };

        document.body.appendChild(element);

        return flushPromises().then(() => {
            // check for updated value on the sge_form-field component
            const value = element.fieldObject;
            expect(value).toMatchSnapshot();
        });
    });

});