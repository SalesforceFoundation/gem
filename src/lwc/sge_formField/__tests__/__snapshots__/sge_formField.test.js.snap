// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-sge_form-field should load with matcing donation info 1`] = `
Object {
  "Test_Number_Field__c": 123,
}
`;

exports[`c-sge_form-field should return a field object with a name and value 1`] = `
Object {
  "Test_Number_Field__c": 10,
}
`;

exports[`c-sge_form-field should work when the value is undefined 1`] = `
Object {
  "Test_Number_Field__c": undefined,
}
`;
